pandas/core/groupby/generic.py

```diff
@@ -1612,27 +1669,63 @@ def _insert_inaxis_grouper_inplace(self, result):
             if in_axis:
                 result.insert(0, name, lev)
 
-    def _wrap_aggregated_output(self, output, names=None):
-        agg_axis = 0 if self.axis == 1 else 1
-        agg_labels = self._obj_with_exclusions._get_axis(agg_axis)
+    def _wrap_aggregated_output(
+        self, output: Mapping[int, AnyArrayLike], columns: Index
+    ) -> DataFrame:
+        """
+        Wraps the output of DataFrameGroupBy aggregations into the expected result.
 
-        output_keys = self._decide_output_index(output, agg_labels)
+        Parameters
+        ----------
+        output : dict[int, np.ndarray]
+            Dict where the key represents the columnar-index and the values are
+            the actual results.
+        columns : pd.Index
+            Column names to apply
+
+        Returns
+        -------
+        DataFrame
+        """
+        result = DataFrame(output)
+        result.columns = columns
```

Comment:
do you need to .sort_index(axis=1)? IOW i don't think the ordering is guaranteed

Reply:
Hmm I might not fully understand what you are asking but the order of output and columns should align internally with how these loops are constructed.

FWIW I think that's a little tenuous long term and should be further cleaned up. We should probably change the keys of `output` to be a NamedTuple that contains those and get rid of storing the labels in a separate container, but leaving to a follow up so as not to change too much in one pass

Reply:
you didnt' answer this here

Reply:
This is what the assert statements are doing - making sure the order of the columns in the output dict is maintained.